include/exadg/operators/integrator_flags.h

```diff
@@ -22,54 +22,66 @@
 #ifndef INCLUDE_EXADG_OPERATORS_INTEGRATOR_FLAGS_H_
 #define INCLUDE_EXADG_OPERATORS_INTEGRATOR_FLAGS_H_
 
+#include <deal.II/matrix_free/evaluation_flags.h>
+
 namespace ExaDG
 {
 struct CellFlags
 {
-  CellFlags(bool const value = false, bool const gradient = false, const bool hessian = false)
-    : value(value), gradient(gradient), hessian(hessian){};
+  explicit CellFlags(
+    dealii::EvaluationFlags::EvaluationFlags const flags_in = dealii::EvaluationFlags::nothing)
+    : flags(flags_in){};
 
   CellFlags
   operator||(CellFlags const & other)
   {
     CellFlags cell_flags_combined;
 
-    cell_flags_combined.value    = this->value || other.value;
-    cell_flags_combined.gradient = this->gradient || other.gradient;
-    cell_flags_combined.hessian  = this->hessian || other.hessian;
+    cell_flags_combined.flags = flags | other.flags;
 
     return cell_flags_combined;
   }
 
-  bool value;
-  bool gradient;
-  bool hessian;
+  // checks weather any of the provided flags is set
+  inline DEAL_II_ALWAYS_INLINE //
+    bool
+    do_eval(dealii::EvaluationFlags::EvaluationFlags other =
+              dealii::EvaluationFlags::values | dealii::EvaluationFlags::gradients |
+              dealii::EvaluationFlags::hessians) const
+  {
+    return other & flags;
+  }
```

Comment:
The default argument is there to be able to call do_eval() without arguments and get the same result as before. That is: any of the operators is evaluated. 

I see that it is not intuitive, I had to think a lot during the implementation as well :D 
Personally, I would rename the function to evaluate_any() which should clear this up. 
Generally, I would stick to the bitwise operations since they are fast and since do_eval() is used inside the kernels. This is also why I added the inline keywords. 

I generally think the design is nice, since it shifts the responsibility for such bitwise operations into a single function. Having multiple functions, e.g. do_values(), do_gradients() just adds lines of code without much benefit.

Reply:
The question is a bit how to proceed, because there were many long discussions on this topic in deal.II (where we discussed all the options with `evaluate_any`, `evaluate_all` and many more), which we probably do not want to repeat here. My wish would be that we do not change any logic in this PR compared to the previous code. The decision made by the deal.II community here: https://github.com/dealii/dealii/pull/13229#issuecomment-1014767396 was to write `If(flags & EvaluationFlags::values)` in the code if you just want to ask for the values. Personally, I did not like this design, but it was a majority decision and now it is what it is. Implying, I would follow the deal.II design for consistency. There would be clean designs in my opinion, but it üprobably does not make much sense to discuss them here, given that I do not see a change to get a majority in deal.II.

Reply:
> So @jh66637 you are suggesting two different functions, one checking if we evaluate at all, and the other doing the bitwise `and` to check whether a particular option is set? I would like that a lot, I think this makes it clearer than trying to use the same function with "funny" default argument.

Kind of. What I had in mind was to rename do_eval() to evaluate_any() and add a second function evaluate_any_of() which requires flags to be passed.  evaluate_any() would then just call evaluate_any_of() with the current default argument of do_eval().

I personally don't like how it would appear in the code if we do the bitwise operation in place, since ExaDG::Flags becomes inconsistent.
```
if(this->integrator_flags.cell_integrate.do_eval()
if(this->integrator_flags.cell_integrate.flags & dealii::EvaluationFlags::values)
```
I would either check the bitwise operation in place everywhere or nowhere. Checking it everywhere is a mess since values|grads|hessians has to be added in a lot of places and if e.g. a new flag is introduces we would have to change a lot of code. Therfore, I would prefer 
```
if(this->integrator_flags.cell_integrate.evaluate_any()
if(this->integrator_flags.cell_integrate.evaluate_any_of(dealii::EvaluationFlags::values)
```

Reply:
I think both options are valid here (the one with bitwise operators and the one with member functions). In the morning, I had a slight preference for the bitwise version because of consistency with how it is done in deal.II, but I am also happy with the version you suggest as a developer. Still, I prefer a version without default argument since this is error-prone in general.